toolbox/api/datagalaxy_api_dictionary.py

```
@@ -0,0 +1,161 @@
+import logging
+
+import requests as requests
+from toolbox.api.datagalaxy_api_workspaces import DataGalaxyBulkResult, to_bulk_properties_tree
+
+
+class DataGalaxyApiDictionary:
+    def __init__(self, url: str, access_token: str, workspace: dict):
+        self.url = url
+        self.access_token = access_token
+        self.workspace = workspace
+
+    def list_sources(self, workspace_name: str) -> list:
+        if not self.workspace["isVersioningEnabled"]:
+            version_id = self.workspace['defaultVersionId']
+            params = {'versionId': version_id, 'includeAttributes': 'true'}
+            headers = {'Authorization': f"Bearer {self.access_token}"}
+            response = requests.get(f"{self.url}/sources", params=params, headers=headers)
+            code = response.status_code
+            body_json = response.json()
+            if code == 200:
+                logging.info(
+                    f'list_sources - {len(body_json["results"])} sources found on '
+                    f'workspace: {workspace_name}')
+                result = []
+                result = result + body_json['results']
+                next_page = body_json["next_page"]
+                while next_page is not None:
+                    headers = {'Authorization': f"Bearer {self.access_token}"}
+                    response = requests.get(next_page, headers=headers)
+                    body_json = response.json()
+                    next_page = body_json["next_page"]
+                    result = result + body_json['results']
+                return result
+
+            if 400 <= code < 500:
+                raise Exception(body_json['error'])
+
+            raise Exception(f'Unexpected error, code: {code}')
```

same here, if `code != 200`, we can raise an exception